### ISSUE TITLE
Not detecting change to defaultProps

### ISSUE THREAD

**Issue by eyal-perry-coralert** [gaearon/react-hot-loader#1047]
### Description
Consider the below component:

export class MyComponent extends React.Component {

 static defaultProps = {
   someKey:'value'
 };

 render(){
  return (
   <div>{this.props.someKey}</div>
  );
 }
}

Try to change the defaultProps value for someKey and see that an update takes place in the browser console, yet no change to the DOM is observed.
I have verified that HMRE is set up correctly for the project, since it generally works.

This may have something to do with babel transpilation.

### Expected behavior

Change should be observed

### Actual behavior

Change is not observed

### Environment

React Hot Loader version: 4.3.4

1.  Node version: 10.8.0
2.  NPM version: 6.1.0
3.   Babel versions:
     "babel-register": "6.26.0",
      "babel-cli": "6.24.1",
      "babel-core": "6.25.0",
      "babel-eslint": "7.2.3",
      "babel-loader": "7.1.1",
      "babel-jest": "22.4.3",
      "babel-preset-env": "1.6.1",
      "babel-preset-react": "6.24.1",
      "babel-preset-react-hmre": "1.1.1",
      "babel-preset-stage-0": "6.24.1",
      "babel-polyfill": "6.26.0",
      "babel-plugin-transform-object-rest-spread": "6.26.0",
      "babel-plugin-transform-async-to-generator": "6.24.1",
4. Webpack versions:
      "webpack": "3.11.0",
      "webpack-dev-server": "2.11.2"

Operating system: Windows 10 x64
Browser and version: Chrome 68.0.3440.106 (Official Build) (64-bit)

### Reproducible Demo
Just copy the above example.

**Comment by theKashey** [comment]
React-Hot-Loader is not changing the past. It preserves the component's past, including component's state, or lifecycle methods.
defaultProps are used in the component constructor, but the component is already constructed.

We thought about better detection of the situations we can't or won't reproduce, and then perform some action, to actually reflect the changes made. But not sure how to do it actually.

**Comment by theKashey** [comment]
Actually - this is something I want to do. Blaming RHL creators, myself included, every time I hit the wall, and have to manually refresh the page.

**Comment by eyal-perry-coralert** [comment]
I guess a simple checksum of module code will do just fine, when defaultProps is defined and not an empty object on the component

**Comment by theKashey** [comment]
Nope. We are here to preserve state (and other information), not to update everything.

**Comment by chuckdries** [comment]
Can we at least show some kind of message to indicate this is the case? I thought I had a mistake in my code, not that the hot loader wasn't working.

**Comment by theKashey** [comment]
RHL does not control, and, from some point of view, could not control defaultProps. Right now the only way is to hook into `createElement` and override the current mechanism, not sure it's a good idea, as long as that moment is about to change - see https://github.com/reactjs/rfcs/pull/107
